exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,585 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
+				wsResponse, err := e.WsCancelTradeOrders(ctx, pairs.Strings(), []AccountType{AccountType(cancelOrd.AssetType)})
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				for _, wco := range wsResponse {
+					if wco.Code == 0 {
+						cancelAllOrdersResponse.Status[strconv.FormatUint(wco.OrderID, 10)] = wco.State
+					}
+				}
+			} else {
+				resp, err := e.CancelTradeOrders(ctx, pairs.Strings(), []AccountType{AccountType(cancelOrd.AssetType)})
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				for _, co := range resp {
+					if co.Code == 0 {
+						cancelAllOrdersResponse.Status[co.OrderID] = co.State
+					}
+				}
+			}
+		}
+	case asset.Futures:
+		result, err := e.CancelFuturesOrders(ctx, cancelOrd.Pair.String(), cancelOrd.Side.String())
+		if err != nil {
+			return cancelAllOrdersResponse, err
 		}
+		for _, co := range result {
+			if co.Code == 200 {
+				cancelAllOrdersResponse.Status[co.OrderID] = "cancelled"
+			}
+		}
+	default:
+		return cancelAllOrdersResponse, fmt.Errorf("%w: %q", asset.ErrNotSupported, cancelOrd.AssetType)
 	}
-
 	return cancelAllOrdersResponse, nil
 }
 
 // GetOrderInfo returns order information based on order ID
-func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, _ asset.Item) (*order.Detail, error) {
+func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, assetType asset.Item) (*order.Detail, error) {
 	if pair.IsEmpty() {
 		return nil, currency.ErrCurrencyPairEmpty
 	}
-
-	orderInfo := order.Detail{
-		Exchange: e.Name,
-		Pair:     pair,
-	}
-
-	trades, err := e.GetAuthenticatedOrderTrades(ctx, orderID)
-	if err != nil && !strings.Contains(err.Error(), "Order not found") {
-		return nil, err
-	}
-
-	for i := range trades {
-		var tradeHistory order.TradeHistory
-		tradeHistory.Exchange = e.Name
-		tradeHistory.Side, err = order.StringToOrderSide(trades[i].Type)
+	switch assetType {
+	case asset.Spot:
+		trades, err := e.GetTradesByOrderID(ctx, orderID)
+		if err != nil {
+			return nil, err
+		}
+		orderTrades := make([]order.TradeHistory, len(trades))
+		for i, td := range trades {
+			var oType order.Type
+			oType, err = order.StringToOrderType(td.Type)
+			if err != nil {
+				return nil, err
+			}
+			orderTrades[i] = order.TradeHistory{
+				Exchange:  e.Name,
+				TID:       td.ID,
+				FeeAsset:  td.FeeCurrency.String(),
+				Price:     td.Price.Float64(),
+				Total:     td.Amount.Float64(),
+				Timestamp: td.CreateTime.Time(),
+				Amount:    td.Quantity.Float64(),
+				Fee:       td.FeeAmount.Float64(),
+				Side:      stringToOrderSide(td.Side),
+				Type:      oType,
+			}
+		}
+		resp, err := e.GetOrder(ctx, orderID, "")
+		if err != nil {
+			smartOrders, err := e.GetSmartOrderDetails(ctx, orderID, "")
+			if err != nil {
+				return nil, err
+			} else if len(smartOrders) == 0 {
+				return nil, order.ErrOrderNotFound
+			}
+			if len(smartOrders) > 0 {
```

Comment:
🐒 Optional nitpick:
We just checked smartOrders len. This'd be cleaner if we just do a var and then simplify everything below.
```
s := smartOrders[0]
```